warzone/Orders.cpp

```diff
@@ -143,11 +140,55 @@ bool AdvanceOrder::execute() {
 		return false;
 	}
 
-	// TODO: Do actions once we have more details.
+	// Actions for when the source and target territories belong to the same player.
+	if (this->source->getOwner() == this->target->getOwner()) {
+		this->source->removeArmies(numArmies);
+		this->target->addArmies(numArmies);
+	}
+	// Execute battle simulation sequence.
+	else {
+		this->battle();
+	}
+
 	cout << "Advance order executed." << endl;
 	return true;
 }
 
+void AdvanceOrder::battle() {
+	// Seed the random number generator.
+	// TODO Probably better if this occurs once at the top of main().
+	srand(time(0));
+
+	while (this->source->getArmies() > 0 && this->target->getArmies() > 0) {
+		// Attacker has 60% chance of killing defender.
+		if ((1 + rand() % 100) < 60) { 
+			this->target->removeArmies(1); 
+		}
+		// Defender has 70% chance of killing attacker.
+		if ((1 + rand() % 100) < 70) {
+			this->source->removeArmies(1);
+		}
+	}
+
+	// When attacker wins, the survivors occupy the territory.
+	if (this->source->getArmies() > this->target->getArmies()) {
+		int numArmies = this->source->getArmies();
+
+		// Place remaining armies on the new territory.
+		this->source->removeArmies(numArmies);
+		this->target->addArmies(numArmies);
+
+		// Transfer ownership of territory to the winning player.
+		this->target->setOwner(this->source->getOwner());
+
+		// Add the new territory to the player's list of territories.
+		this->player->addTerritory(this->target);
```

Comment:
First you have to remove the territory from the defending player's list of territories.